h/views/api/group_members.py

```diff
@@ -54,3 +60,33 @@ def add_member(context: GroupMembershipContext, request):
     group_members_service.member_join(context.group, context.user.userid)
 
     return HTTPNoContent()
+
+
+@api_config(
+    versions=["v1", "v2"],
+    route_name="api.group_member",
+    request_method="PATCH",
+    link_name="group.member.edit",
+    description="Change a user's role in a group",
+)
+def edit_member(context: GroupMembershipContext, request):
+    appstruct = EditGroupMembershipAPISchema().validate(json_payload(request))
+    old_roles = context.membership.roles
+    new_roles = appstruct["roles"]
+
+    if not request.has_permission(
+        Permission.Group.MEMBER_EDIT,
+        EditGroupMembershipContext(
+            context.group, context.user, context.membership, new_roles
+        ),
+    ):
+        raise HTTPNotFound()
+
+    context.membership.roles = new_roles
```

Comment:
There's no need for a service to carry out the business logic, it's just an assignment.